### PR TITLE
Add "leading sponsor" setup to sponsors display, fixes #44

### LINKED ISSUE #44: Add key sponsors (GitHub or not) to support page
We have the key leading sponsors on the GitHub README

image

but we should probably try to get them onto the [ddev.com support page](https://ddev.com/support-ddev/)

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

* #44 

Platform.sh and future "lead sponsors" want a more prominent listing.

Review at https://20241108-add-leading-sponsor.ddev-com-front-end.pages.dev/#supporters

## Manual Testing Instructions

Review the main page in regular and dark mode.

### CODE REVIEW

**Review by stasadev — CHANGES_REQUESTED** [review]
https://20241108-add-leading-sponsor.ddev-com-front-end.pages.dev/#supporters

Light, I see the vertical line:

image

Dark, I don't see the vertical line (if it matters):

image

---

Testing the direct link with

https://github.com/stasadev/ddev/tree/20241111_stasadev_featured_sponsors?tab=readme-ov-file#wonderful-sponsors

Light (good):

https://20241108-add-leading-sponsor.ddev-com-front-end.pages.dev/resources/featured-sponsors.svg

image

Dark (shows nothing for me):

https://20241108-add-leading-sponsor.ddev-com-front-end.pages.dev/resources/featured-sponsors-darkmode.svg

image

**Comment by bmartinez287** [comment]
Screenshot 2024-11-11 at 6 42 11 PM
`dark:border-white`
For dark mode we probably want the border white.

File: src/components/FeaturedSponsors.astro [hunk 1]
```diff
@@ -19,41 +18,69 @@ const nudges = {
...
     sponsor.class = nudges[sponsor.name]
   }
 })
 ---
 
-<div class="relative">
+<div class="relative flex flex-col items-center space-y-12">
+  {/* Lead Sponsors Section */}
+  {leadSponsors.length > 0 && (
+      <div class="w-full flex justify-center pb-8 border-b border-gray-200 dark:border-gray-800">
```
Comment (bmartinez287) [inline src/components/FeaturedSponsors.astro:36]:
```suggestion
      <div class="w-full flex justify-center pb-8 border-b border-gray-200 dark:[border-gray-800](dark:border-white)"> 
```

**Comment by rfay** [comment]
OK, trying it out, thanks!

**Review by bmartinez287 — APPROVED** [review]

File: src/pages/resources/featured-sponsors-darkmode.svg.js [hunk 2]
```diff
@@ -7,54 +7,128 @@ const baseUrl = import.meta.env.SITE
...
+  const regularSponsors = featuredSponsors.filter(sponsor => sponsor.isLeading !== true)
+
+  console.log("Processing lead sponsors")
+
+  // Handle lead sponsors (centered, own row)
+  if (leadSponsors.length > 0) {
+    // Get dimensions for lead sponsors
+    const leadSponsorInfo = leadSponsors.map(sponsor => {
+      // Ensure we have a valid logo path
+      const logoPath = sponsor.darkLogo ? `./public${sponsor.darkLogo}` : null
```
Comment (stasadev) [inline src/pages/resources/featured-sponsors-darkmode.svg.js:40]:
```suggestion
      const logoPath = sponsor.darklogo ? `./public${sponsor.darklogo}` : null
```

File: src/pages/resources/featured-sponsors-darkmode.svg.js [hunk 3]
```diff
@@ -7,54 +7,128 @@ const baseUrl = import.meta.env.SITE
...
+      return sum + info.width + (index < leadSponsors.length - 1 ? xPadding : 0)
+    }, 0)
+
+    // Center the lead sponsors
+    let startX = (overallWidth - totalLeadWidth) / 2
+
+    // Place lead sponsors
+    leadSponsorInfo.forEach(({ sponsor, width, height, logoPath }) => {
+      images.push({
+        href: baseUrl + sponsor.darkLogo,
```
Comment (stasadev) [inline src/pages/resources/featured-sponsors-darkmode.svg.js:71]:
```suggestion
        href: baseUrl + sponsor.darklogo,
```

File: src/pages/resources/featured-sponsors-darkmode.svg.js [hunk 4]
```diff
@@ -7,54 +7,128 @@ const baseUrl = import.meta.env.SITE
...
 
-    if (currentX + w + xPadding > overallWidth) {
-      currentX = 0
-      currentY += maxHeight + yPadding
-      rowCount += 1
+  // Now process regular sponsors
+  let currentX = 0
+  regularSponsors.forEach((sponsor) => {
+    // Ensure we have a valid logo path
+    const logoPath = sponsor.darkLogo ? `./public${sponsor.darkLogo}` : null
```
Comment (stasadev) [inline src/pages/resources/featured-sponsors-darkmode.svg.js:91]:
```suggestion
    const logoPath = sponsor.darklogo ? `./public${sponsor.darklogo}` : null
```

File: src/pages/resources/featured-sponsors-darkmode.svg.js [hunk 5]
```diff
@@ -7,54 +7,128 @@ const baseUrl = import.meta.env.SITE
...
+      )
+
+      // Start new row if needed
+      if (currentX + width > overallWidth) {
+        currentX = 0
+        currentY += maxHeight + yPadding
+      }
+
+      images.push({
+        href: baseUrl + sponsor.darkLogo,
```
Comment (stasadev) [inline src/pages/resources/featured-sponsors-darkmode.svg.js:112]:
```suggestion
        href: baseUrl + sponsor.darklogo,
```

**Review by stasadev — CHANGES_REQUESTED** [review]
Suggestion to fix the empty SVG:

https://20241108-add-leading-sponsor.ddev-com-front-end.pages.dev/resources/featured-sponsors-darkmode.svg

File: src/pages/resources/featured-sponsors-darkmode.svg.js (hunk 2 above)
Comment (rfay) [reply src/pages/resources/featured-sponsors-darkmode.svg.js:40]: Oh dear. Thanks. Caps should be outlawed.

**Pull request merged**